FeedStoreChallenge/ObjectBoxFeedStore.swift

```diff
@@ -0,0 +1,161 @@
+//
+//  ObjectBoxFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by Riccardo Rossi - Home on 22/03/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import Foundation
+import ObjectBox
+
+typealias StoreFeed = ObjectBoxFeedStore.StoreFeed
+typealias Cache = ObjectBoxFeedStore.Cache
+
+public class ObjectBoxFeedStore: FeedStore {
+	
+	// MARK: - Inner classes & enums
+	
+	class StoreFeed: Entity {
+		var id: Id = 0
+		var modelId: String = ""
+		var description: String?
+		var location: String?
+		var url: String = ""
+		var cache: ToOne<Cache> = nil
+		
+		init(modelId: String = "", description: String? = nil, location: String? = nil, url: String = "", cache: Cache? = nil) {
+			self.modelId = modelId
+			self.description = description
+			self.location = location
+			self.url = url
+			self.cache.target = cache
```

Comment:
Ideally, you should define an interface that prevents clients from creating the model with the wrong state. For example, `modelId` shouldn't be *any* `String` in the initializer. It should be a UUID. And the url should be a `URL`.

And they shouldn't have default values.

```suggestion
		init(modelId: UUID, description: String?, location: String?, url: URL, cache: Cache) {
			self.modelId = modelId.uuidString
			self.description = description
			self.location = location
			self.url = url.absoluteString
			self.cache.target = cache
```

Reply:
Right! Just one consideration: I had to add default parameters values to satisfy the following code generated from a ObjectBox's script `generated/EntityInfo-FeedStoreChallenge.generated.swift`:

`... let entity = StoreFeed() ...`

For this reason i created an initializer how you correctly pointed out and one to satisfy the previous one:

`init() {
	self.modelId = ""
	self.description = nil
	self.location = nil
	self.url = ""
	self.cache = nil
}`

Reply:
Hi @RRReDz, I meant it shouldn't have default values for the params in the initializer. But the properties can have default values to satisfy the framework requirements.